onadata/apps/api/tests/viewsets/test_attachment_viewset.py

```diff
@@ -229,8 +229,10 @@ def test_list_view_surfaces_current_attachment(self):
         self.assertEqual(len(response.data), 5)
 
         # test when the attachment is soft deleted
-        self.attachment.soft_delete()
+        self.attachment.soft_delete(user=self.user)
         self.attachment.save()
+        # Test that deleted_by field captures the right user
+        self.assertTrue(self.attachment.deleted_by, self.user)
```

Comment:
Can we move this to a separate test please?

So not assign a user here, add a new independent test that assigns user and tests they are assigned to delete object

Having each test only test for one thing is ideal

Reply:
sure. Let me move that.